src/transformers/models/zoedepth/modeling_zoedepth.py

```diff
@@ -0,0 +1,1349 @@
+            for i in range(self.n_attractors):
+                # shape batch_size, num_bins, height, width
+                delta_c += dist(attractors[:, i, ...].unsqueeze(1) - bin_centers)
+
+            if self.kind == "mean":
+                delta_c = delta_c / self.n_attractors
+
+        bin_new_centers = bin_centers + delta_c
+        bin_centers = bin_new_centers
+
+        return bin_new_centers, bin_centers
+
+
+class ZoeDepthProjector(nn.Module):
+    def __init__(self, in_features, out_features, mlp_dim=128):
+        """Projector MLP.
+
+        Args:
+            in_features (`int`):
+                Number of input channels.
+            out_features (`int`):
+                Number of output channels.
+            mlp_dim (`int`, *optional*, defaults to 128):
+                Hidden dimension.
+        """
+        super().__init__()
+
+        self.conv1 = nn.Conv2d(in_features, mlp_dim, 1, 1, 0)
+        self.act = nn.ReLU(inplace=True)
+        self.conv2 = nn.Conv2d(mlp_dim, out_features, 1, 1, 0)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.act(x)
+        x = self.conv2(x)
+
+        return x
+
+
+class ZoeDepthPatchTransformerEncoder(nn.Module):
+    def __init__(self, in_channels, patch_size=10, embedding_dim=128, num_heads=4, use_class_token=False):
+        """ViT-like transformer block
+
+        Args:
+            in_channels (`int`):
+                Input channels.
+            patch_size (`int`, *optional*, defaults to 10):
+                Patch size.
+            embedding_dim (`int`, *optional*, defaults to 128):
+                Embedding dimension in transformer model.
+            num_heads (`int`, *optional*, defaults to 4):
+                Number of attention heads.
+            use_class_token (`bool`, *optional*, defaults to `False`):
+                Whether to use extra token at the start for global accumulation (called as "class token").
+        """
+        super().__init__()
+        encoder_layers = nn.TransformerEncoderLayer(embedding_dim, num_heads, dim_feedforward=1024, batch_first=True)
+        self.transformer_encoder = nn.TransformerEncoder(encoder_layers, num_layers=4)
+        self.use_class_token = use_class_token
+
+        self.embedding_convPxP = nn.Conv2d(
+            in_channels, embedding_dim, kernel_size=patch_size, stride=patch_size, padding=0
+        )
+
+    def positional_encoding_1d(self, batch_size, sequence_length, embedding_dim, device="cpu"):
+        """Generate positional encodings
+
+        Args:
+            sequence_length (int): Sequence length
+            embedding_dim (int): Embedding dimension
+
+        Returns:
+            torch.Tensor: Positional encodings.
+        """
+        position = torch.arange(0, sequence_length, dtype=torch.float32, device=device).unsqueeze(1)
+        index = torch.arange(0, embedding_dim, 2, dtype=torch.float32, device=device).unsqueeze(0)
```

Comment:
Is having float32 here compatible with loading the model in and using fp16 inputs?